Code/Tools/AssetProcessor/native/utilities/SimpleStatsCapture.cpp

```diff
@@ -0,0 +1,392 @@
+            void Dump();
+        private:
+            using timepoint = AZStd::chrono::high_resolution_clock::time_point;
+            using duration = AZStd::chrono::milliseconds;
+            struct StatsEntry
+            {
+                duration m_cumulativeTime = {};    // The total amount of time spent on this.
+                timepoint m_operationStartTime = {}; // Async tracking - the last time stamp an operation started.
+                int64_t m_operationCount = 0; // In case there's more than one sample.  Used to calc average.
+            };
+            
+            AZStd::unordered_map<AZStd::string, StatsEntry> m_stats;
+            // utility function - pushes all keys in m_stats into a vector
+            void PushKeys(AZStd::vector<AZStd::string>& dest)
+            {
+                for (const auto& element : m_stats)
+                {
+                    dest.push_back(element.first);
+                }
+            }
+            
+            // Make a friendly time string of the format nnHnnMhhS.xxxms
+            AZStd::string FormatDuration(const duration& duration)
+            {
+                int64_t milliseconds = duration.count();
+                constexpr int64_t millisecondsInASecond = 1000;
+                constexpr int64_t millisecondsInAMinute = millisecondsInASecond * 60;
+                constexpr int64_t millisecondsInAnHour = millisecondsInAMinute * 60;
+                
+                int64_t hours = milliseconds / millisecondsInAnHour;
+                milliseconds -= hours * millisecondsInAnHour;
+                
+                int64_t minutes = milliseconds / millisecondsInAMinute;
+                milliseconds -= minutes * millisecondsInAMinute;
+                
+                int64_t seconds = milliseconds / millisecondsInASecond;
+                milliseconds -= seconds * millisecondsInASecond;
+                
+                // omit the sections which dont make sense for readability
+                if (hours)
+                {
+                    return AZStd::string::format("%02" PRId64 "h%02" PRId64 "m%02" PRId64 "s%03" PRId64 "ms" , hours, minutes, seconds, milliseconds);
+                }
+                else if (minutes)
+                {
+                    return AZStd::string::format("   %02" PRId64 "m%02" PRId64 "s%03" PRId64 "ms", minutes, seconds, milliseconds);
+                }
+                else if (seconds)
+                {
+                    return AZStd::string::format("      %02" PRId64 "s%03" PRId64 "ms", seconds, milliseconds);
+                }
+                
+                return AZStd::string::format("         %03" PRId64 "ms", milliseconds);
+            }
+        };
+
+        void SimpleStatsCaptureImpl::BeginCaptureStat(AZStd::string_view statName)
+        {
+            StatsEntry& existingStat = m_stats[statName];
+            if (existingStat.m_operationStartTime != timepoint())
```

Comment:
Depending on how precise the timer is, this may need a window of a few nanoseconds to compensate for function call overhead.

Reply:
The events we're capturing here are still in the seconds, minutes range.  But we are using chrono:: milliseconds high resolution so it should be ok.